docs/README.md

```diff
@@ -0,0 +1,320 @@
+-   Check for [known issues](https://github.com/aspect-build/rules_js/issues)
+-   Pay for support, provided by <https://aspect.dev>.
+
+## Installation
+
+From the release you wish to use:
+<https://github.com/aspect-build/rules_js/releases>
+copy the WORKSPACE snippet into your `WORKSPACE` file.
+
+Note that [bzlmod] is experimentally supported as well.
+
+## Usage
+
+### Bazel basics
+
+Bazel's `BUILD` or `BUILD.bazel` files are used to declare the dependency graph of your code.
+They describe the source files and their dependencies, and declare entry points for programs or tests.
+However, they don't say _how to build_ the code, that's the job of Bazel rules.
+
+Because `BUILD` files typically declare a finer-grained dependency graph than `package.json` files, Bazel can be smarter about what to fetch or invalidate for a given build.
+For example, Bazel might only need to fetch a single npm package for a simple build,
+where you might experience other tools installing the entire `package.json` file.
+
+On the other hand, authoring BUILD files is a chore that's required under Bazel and not in
+other tools like Rush or Nx. We plan a [Gazelle] extension soon which automates much of this toil.
+
+Other recommendations:
+
+-   Put [common flags](https://blog.aspect.dev/bazelrc-flags) in your `.bazelrc` file.
+-   Use [Renovate](https://docs.renovatebot.com/) to keep your Bazel dependencies up-to-date.
+
+### Node.js
+
+rules_js depends on rules_nodejs version 5.0 or greater.
+
+Installation is included in the `WORKSPACE` snippet you pasted from the Installation instructions above.
+
+**API docs:**
+
+-   Installing and choosing the version of Node.js:
+    <https://bazelbuild.github.io/rules_nodejs/install.html>
+-   Rules API: <https://bazelbuild.github.io/rules_nodejs/Core.html>
+-   The Node.js toolchain: <https://bazelbuild.github.io/rules_nodejs/Toolchains.html>
+
+### Use third-party packages from npm
+
+rules_js accesses npm packages using [pnpm].
+pnpm's "virtual store" of packages aligns with Bazel's "external repositories",
+and the pnpm "linker" which creates the `node_modules` tree has semantics we can reproduce with Bazel actions.
+
+If your code works with pnpm, then you should expect it works under Bazel as well.
+This means that if your issue can be reproduced outside of Bazel, using a reproduction with only pnpm,
+then we ask that you fix the issue there, and will close such issues filed on rules_js.
+
+The typical usage is to import an entire `pnpm-lock.yaml` file.
+Create such a file if you don't have one. You could install pnpm on your machine, or use `npx` to run it.
+For example, this command creates a lockfile with minimal installation needed:
+
+```shell
+$ npx pnpm install --lockfile-only
+```
+
+Next, you'll typically use `npm_translate_lock` to translate the Yaml file to Starlark, which Bazel extensions understand.
+The `WORKSPACE` snippet you pasted above already contains this code.
+
```

Comment:
Should we post two screenshots or code here to display the `pnpm lock` and the `Starlark` version side by side, it may be eaiser for the user to understand.

Reply:
I don't think users would understand the content of either file actually. They're both opaque blobs you never read, and GitHub and other code review tools don't even show you the changes. I did add a section here about how you could check in the generated file though, rather than compute on-the-fly. Do you think it's understandable?